pyanaconda/ui/gui/spokes/root_password.py

```diff
@@ -51,51 +56,72 @@ class PasswordSpoke(FirstbootSpokeMixIn, NormalSpoke, GUISpokeInputCheckHandler)
-        self._pwASCIICheck = self.add_check(self.pw, self.check_password_ASCII)
 
-        self._kickstarted = self.data.rootpw.seen
-        if self._kickstarted:
-            self.pw.set_placeholder_text(_("The password is set."))
-            self.confirm.set_placeholder_text(_("The password is set."))
-
-        self.pw_bar = self.builder.get_object("password_bar")
-        self.pw_label = self.builder.get_object("password_label")
+        # Setup the password checker for password checking
+        self._checker = input_checking.PasswordChecker(initial_password_content = self.password,
+                                                       initial_password_confirmation_content = self.password_confirmation,
+                                                       policy = input_checking.get_policy(self.data, "root"))
+        # configure the checker for password checking
+        self.checker.name_of_password = _(constants.NAME_OF_PASSWORD)
+        self.checker.name_of_password_plural = _(constants.NAME_OF_PASSWORD_PLURAL)
+        # remove any placeholder texts if either password or confirmation field changes content from initial state
+        self.checker.password.changed_from_initial_state.connect(self.remove_placeholder_texts)
+        self.checker.password_confirmation.changed_from_initial_state.connect(self.remove_placeholder_texts)
+        # connect UI updates to check results
+        self.checker.checks_done.connect(self._checks_done)
+
+        # check that the password is not empty
+        self._empty_check = input_checking.PasswordEmptyCheck()
+        # check that the content of the password field & the conformation field are the same
+        self._confirm_check = input_checking.PasswordConfirmationCheck()
+        # regards both fields empty as success to let the user escape
+        self._confirm_check.success_if_confirmation_empty = True
+        # check password validity, quality and strength
+        self._validity_check = input_checking.PasswordValidityCheck()
+        # connect UI updates to validity check results
+        self._validity_check.result.password_score_changed.connect(self.set_password_score)
+        self._validity_check.result.status_text_changed.connect(self.set_password_status)
+        # check if the password contains non-ascii characters
+        self._ascii_check = input_checking.PasswordASCIICheck()
+
+        # register the individual checks with the checker in proper order
+        # 1) is the password non-empty ?
+        # 2) are both entered passwords the same ?
+        # 3) is the password valid according to the current password checking policy ?
+        # 4) is the password free of non-ASCII characters ?
+        self.checker.add_check(self._empty_check)
+        self.checker.add_check(self._confirm_check)
+        self.checker.add_check(self._validity_check)
+        self.checker.add_check(self._ascii_check)
```

Comment:
I would like to see all the checkers setup in one method.

Reply:
That's not really possible - the checks not the same between the root and user spokes and you also need to have a reference for them to access them later on. I guess a separate spoke specific "setup checks" method might work though.